app/models/post.rb

```diff
@@ -0,0 +1,29 @@
+class Post < ActiveRecord::Base
+  # title should be required & unique
+  validates :title, presence: true, uniqueness: true, length: {minimum: 7}
+  validates :body,  presence: true
+
+  has_many   :comments, dependent: :nullify
+  belongs_to :category
+  belongs_to :user
+
+  has_many    :favourites, dependent: :nullify
+  has_many    :favouriting_users, through: :favourites, source: :users
+
```

Comment:
I think it's better to have validates after the relationship. Just my preference.